### PR TITLE
feature: Confirmation Dialog

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR updates the browsers default confirm dialog with a custom html dialog element as demonstrated in [Go Rails - Custom Turbo Confirm Modals with Hotwire](https://gorails.com/episodes/custom-hotwire-turbo-confirm-modals).

**Summary**: Confirm dialog is overwritten in `app/javascript/application.js` with custom overrides provided by data-attributes.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

1. Default hotwire confirm message:
image

2. Custom html content:
image

3. Custom form:
image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

All states can be previews via Lookbook Previews:

1. [Default](http://localhost:3000/rails/lookbook/inspect/confirmation/default)
2. [With custom HTML](http://localhost:3000/rails/lookbook/inspect/confirmation/custom_content)
3. [With custom form](http://localhost:3000/rails/lookbook/inspect/confirmation/with_custom_form)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated
  the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: test/system/projects/members_test.rb [hunk 1]
```diff
@@ -49,9 +49,10 @@ def setup
       visit namespace_project_members_url(@namespace, @project)
 
       all('.member-settings-ellipsis')[2].click
+      click_link I18n.t(:'projects.members.index.remove')
 
-      accept_confirm do
-        click_link I18n.t(:'projects.members.index.remove')
+      within('#turbo-confirm[open]') do
+        click_button 'Confirm'
```
Comment (deepsidhu85) [inline test/system/projects/members_test.rb:55]: Internationalize text here and below

File: test/system/projects/samples_test.rb [hunk 2]
```diff
@@ -60,8 +60,10 @@ class SamplesTest < ApplicationSystemTestCase
     test 'should destroy Sample' do
       visit namespace_project_sample_url(namespace_id: @namespace.path, project_id: @project.path, id: @sample1.id)
       assert_selector 'a', text: I18n.t('projects.samples.index.remove_button'), count: 1
-      accept_confirm do
-        click_link I18n.t('projects.samples.index.remove_button'), match: :first
+      click_link I18n.t(:'projects.samples.index.remove_button')
+
+      within('#turbo-confirm[open]') do
+        click_button 'Confirm'
```
Comment (deepsidhu85) [inline test/system/projects/samples_test.rb:66]: Internationalize text

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks good to me. Just a few internationalizations

File: test/system/projects/samples_test.rb (hunk 2 above)
Comment (joshsadam) [reply test/system/projects/samples_test.rb]: Thanks, updated in 5b27f0b

File: test/system/projects/members_test.rb (hunk 1 above)
Comment (joshsadam) [reply test/system/projects/members_test.rb]: Thanks, updated in 5b27f0b

**Review by deepsidhu85 — APPROVED** [review]

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**